src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -442,6 +442,24 @@ func (f *FilesystemShare) shareRootFilesystemWithNydus(ctx context.Context, c *C
 		guestPath: rootfsGuestPath,
 	}, nil
 }
+func (f *FilesystemShare) shareRootFilesystemWithImagePull(ctx context.Context, c *Container) (*SharedFile, error) {
+	rootfs := &grpc.Storage{}
+	rootfs.MountPoint = filepath.Join("/run/kata-containers/", c.id, c.rootfsSuffix)
```

Comment:
Can I ask for a bit of clarity here. Is the way this is working that `/run/kata-containers/` from the host is mounted to the guest and then used for pull the image too, or something else?

Reply:
@stevenhorsman  No, that simply sets the rootfs mountpoint for the container within the guest, which will subsequently perform a bind mount of the directory fetched by the image_rpc to this mountpoint.